java/code/src/com/suse/manager/matcher/MatcherJsonIO.java

```diff
@@ -181,6 +182,7 @@ public MatcherJsonIO() {
     private static Set<Long> getVirtualGuests(Server system) {
         return system.getGuests().stream()
             .filter(vi -> vi.getGuestSystem() != null)
+            .filter(vi -> !vi.getPayg())
```

Comment:
Hmm, We need to check if a PAYG subscription really obsolete all subscriptions.
I could imagin that we do not need a SLES subscription, but what about a SUSE Manager Lifecycle Subscription?
@joesusecom ?

Reply:
CC @hustodemon @moio 

I talked to @BLandorf and this is a problem. None of the Payg instances has SUSE Manager Lifecycle Subscription included. If we would just remove them from the list we would do something wrong.

I think we should list the system, but without "product ids". When we prepare the Json for the matcher we just add the Lifecycle product ids.

Could this work?

Reply:
I've pushed the changes needed here.